doc/api/next_api_changes/development/26621-ES.rst

```diff
@@ -0,0 +1,43 @@
+Build system ported to Meson
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The build system of Matplotlib has been ported from setuptools to `meson-python
+<https://meson-python.readthedocs.io>`_ and `Meson <https://mesonbuild.com>`_.
+Consequently, there has been a few changes development and packaging purposes.
+
+1. Installation by ``pip`` of packages with ``pyproject.toml`` use `build isolation
```

Comment:
I went down the rabbit hole reading this thread:
https://github.com/pypa/pip/issues/11440
I was hoping that I could `pip install -e .[dev]` to get the build dependencies that you put into optional dev, but that doesn't appear to be the case. So is the [dev] dependency group even useful currently?

Giving users some quick commands to install build dependencies would be nice, so perhaps adding those as a requirements-dev.txt file instead and indicating we need to install from there first?

Reply:
Yea, it does seem like a problem as the extra's dependencies are installed after the package itself is built. Putting it in a `requirements-dev.txt` might be the most convenient.